torch_glow/src/PyTorchModelLoader.h

```diff
@@ -33,99 +34,130 @@ class PyTorchModelLoader {
   /// Defines type for mapping Symbols to PyTorchModelLoader member functions
   /// for loading torch::jit::Node * objects.
   using MappingOfMemberFunctions =
-      std::unordered_map<torch::jit::Symbol, void (PyTorchModelLoader::*)(
-                                                 const torch::jit::Node *)>;
+      std::unordered_map<torch::jit::Symbol,
+                         llvm::Error (PyTorchModelLoader::*)(
+                             const torch::jit::Node *)>;
 
 public:
-  /// Takes a glow::Function \p F, a jit::Graph \p subgraph to load, and a
-  /// stack of \p inputs for the subgraph to be loaded. Output parameters
-  /// \p inputPlaceholders and \p outputPlaceholders are filled out.
-  PyTorchModelLoader(glow::Function &F, torch::jit::Graph &subgraph,
-                     at::ArrayRef<torch::jit::IValue> &inputs,
-                     std::vector<glow::Placeholder *> &inputPlaceholders,
-                     std::vector<glow::Placeholder *> &outputPlaceholders);
-
   /// Returns whether or not a PyTorch node is supported.
   /// NOTE: For now this is just an enumeration of all type of PyTorch nodes
   /// that the loader knows about but doesn't really guarantee that loading
   /// will succeed because determining this requires more informations such as
   /// shape info that isn't yet available when this is run.
   static bool isNodeSupported(const torch::jit::Node *node);
 
+  /// Takes a glow::Function \p F, a jit::Graph \p subgraph to load, and a
+  /// stack of \p inputs for the subgraph to be loaded. Output parameters
+  /// \p inputPlaceholders and \p outputPlaceholders are filled out.
+  /// \returns error on failure
+  static llvm::Error
+  loadJITGraph(glow::Function &F, torch::jit::Graph &subgraph,
```

Comment:
Looking good. 
What are the different use cases for loading PT-JIT subgraph using the CTOR and using this static method?

Reply:
@mortzur the constructor does all of the loading which means it can fail but constructors can't return anything so we have to pass in an llvm::Error that gets assigned to if any error occurs. In order to avoid all of the unpleasantness that this causes, this makes the constructor private and just calls it from from loadJITGraph which aslo manages the error creation and returning. Best of both worlds (functional-style errors, stateful class for building graph)